src/Elastic.Apm/Metrics/MetricsProvider/SystemTotalCpuProvider.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Elastic.Apm.Api;
+using Elastic.Apm.Logging;
+
+namespace Elastic.Apm.Metrics.MetricsProvider
+{
+	public class SystemTotalCpuProvider : IMetricsProvider
+	{
+		private const string SystemCpuTotalPct = "system.cpu.total.norm.pct";
+		private readonly IApmLogger _logger;
+
+		public SystemTotalCpuProvider(IApmLogger logger) => _logger = logger.Scoped(nameof(SystemTotalCpuProvider));
+
+		private bool _firstTotal = true;
+
+		private bool _getTotalProcessorTimeFailedLog;
+		private TimeSpan _lastCurrentTotalProcessCpuTime;
+		private DateTime _lastTotalTick;
+		private Version _processAssemblyVersion;
+		private PerformanceCounter _processorTimePerfCounter;
+
+		public int ConsecutiveNumberOfFailedReads { get; set; }
+		public string NameInLogs => "total system CPU time";
+
+		public IEnumerable<Sample> GetValue()
+		{
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I would split Windows and non-Windows part into separate classes.

Reply:
Addressed in https://github.com/elastic/apm-agent-dotnet/pull/248#discussion_r291309760